actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -84,7 +84,45 @@ def formats
         end
       end
 
-      # Sets the \variant for template.
+      # Sets the \variant for the response template.
+      #
+      # Action View will incorporate the request's variants when determining which template to render. For example, if an <tt>ArticlesController</tt> needs to respond to an <tt>#index</tt> action request with <tt>request.variant = [:ios, :turbo_native]</tt>, it will render the first template file it can find in the following list:
+      #
+      # - `app/views/articles/index.html+ios.erb`
+      # - `app/views/articles/index.html+turbo_native.erb`
+      # - `app/views/articles/index.html.erb`
+      #
+      # Variant names can refer to any aspect of a request, from the type of
+      # User Agent (e.g. iOS, Android, etc.), to the version of the requesting
+      # application (e.g. v1, v2, etc.) to the level of permissions available to
+      # the requesting User (e.g. administrator, contributor, guest, etc.).
+      #
+      # A word of caution: be stingy when adding new variants. Only reach for
+      # variants when groups of conditionals (that share a theme) or other
+      # more traditional templating techniques become unmanageable.
+      #
+      # ==== Parameters
+      #
+      # * +variant+ - a Symbol name or an Array of Symbol names for variants
+      #   used to render the response template
+      #
+      # ==== Examples
+      #
+      #   class ApplicationController < ActionController::Base
+      #     before_action :determine_variants
+      #
+      #     private
+      #       def determine_variants
+      #         variants = []
+      #
+      #         # some code to determine the variant(s) to use
+      #
+      #         variants << :ios if request.user_agent.include? "iOS"
+      #         variants << :turbo_native if request.user_agent.include? "Turbo Native"
```

Comment:
```suggestion
      #         variants << :ios if request.user_agent.include?("iOS")
      #         variants << :turbo_native if request.user_agent.include?("Turbo Native")
```

Or, whichever passes our rubocop, which doesn't currently check code examples in the RDoc.

I just started adding these recently out of habit because `prism` was having trouble parsing seattlerb style. That's probably fixed now though.